src/bokeh/util/serialization.py

```diff
@@ -52,19 +56,20 @@
 #-----------------------------------------------------------------------------
 
 @lru_cache(None)
-def _compute_datetime_types() -> set[type]:
-    import pandas as pd
+def _compute_datetime_types(pandas_imported: bool) -> set[type]:
 
     result = {dt.time, dt.datetime, np.datetime64}
-    result.add(pd.Timestamp)
-    result.add(pd.Timedelta)
-    result.add(pd.Period)
-    result.add(type(pd.NaT))
+    if pandas_imported:
+        import pandas as pd
+        result.add(pd.Timestamp)
+        result.add(pd.Timedelta)
+        result.add(pd.Period)
+        result.add(type(pd.NaT))
     return result
 
 def __getattr__(name: str) -> Any:
     if name == "DATETIME_TYPES":
-        return _compute_datetime_types()
+        return _compute_datetime_types(_is_installed("pandas") and "pandas" in sys.modules)
```

Comment:
We could also just use `sys.modules`.

```suggestion
        return _compute_datetime_types("pandas" in sys.modules)
```

Reply:
I don't see why this has to be an argument to this function altogether. I would just compute this boolean within this function.

Reply:
`_compute_datetime_types` is decorated with `lru_cache`